### PR TITLE
remove the non-generated client

### PR DESCRIPTION
**Pull request by deads2k**
Removes the non-generated client from kube.  The package has a few methods left, but nothing that needs updating when adding new groups.

@ingvagabund

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35431)

### CODE REVIEW

File: pkg/client/typed/discovery/helper.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/version"
+	// Import solely to initialize client auth plugins.
+	_ "k8s.io/kubernetes/plugin/pkg/client/auth"
+)
+
+// MatchesServerVersion queries the server to compares the build version
```
Comment (sttts) [inline pkg/client/typed/discovery/helper.go]: s/compares/compare/

File: pkg/client/typed/discovery/helper.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	if client == nil {
+		client, err = NewDiscoveryClientForConfig(c)
+		if err != nil {
+			return err
+		}
+	}
+	cVer := version.Get()
+	sVer, err := client.ServerVersion()
+	if err != nil {
+		return fmt.Errorf("couldn't read version from server: %v\n", err)
```
Comment (sttts) [inline pkg/client/typed/discovery/helper.go]: no `\n`

File: pkg/client/typed/discovery/helper.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/version"
+	// Import solely to initialize client auth plugins.
+	_ "k8s.io/kubernetes/plugin/pkg/client/auth"
+)
+
+// MatchesServerVersion queries the server to compares the build version
+// (git hash) of the client with the server's build version. It returns an error
+// if it failed to contact the server or if the versions are not an exact match.
+func MatchesServerVersion(client DiscoveryInterface, c *restclient.Config) error {
```
Comment (sttts) [inline pkg/client/typed/discovery/helper.go]: `(bool, error)` is the canonical return type here.

File: pkg/client/typed/discovery/helper.go (hunk 1 above)
Comment (sttts) [inline pkg/client/typed/discovery/helper.go]: Just noticed, it's an old function moved here.

File: pkg/client/typed/discovery/helper.go (hunk 2 above)
Comment (deads2k) [inline pkg/client/typed/discovery/helper.go]: Well, straight move minus the, "change the types to accept what you mean".

**Comment by sttts** [comment]
Please split the file removal from the changes in the second commit.

`ClientCache.discoveryClient` is never written, only read. Would be worth to move ugly initialization in MatchesServerVersion to the ClientCache initialization. Or at least add a TODO.

**Comment by sttts** [comment]
Otherwise, all the changes look mechanical. lgtm. You will win the most-lines-removed award for 1.5 with this.

**Comment by ingvagabund** [comment]
The helper's ``MatchesServerVersion`` and ``NegotiateVersion`` are perfect candidates for moving under ``discovery``. Looks good.

**Comment by caesarxuchao** [comment]
Good job!

**Pull request merged**